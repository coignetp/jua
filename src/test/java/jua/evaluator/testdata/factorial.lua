-- defines a factorial function
function fact (n)
  if n == 0 then
    return 1
  else
    return n * fact(n-1)
  end
end

print(fact(5))
print(fact(10))
-- Should ignore the print(2) parameter but still execute it
print(fact(10, print(2)))
