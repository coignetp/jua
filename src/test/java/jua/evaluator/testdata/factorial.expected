120
3628800
2
3628800
